### ISSUE TITLE
`equatable_if_let` should not expand macros in suggestion

### ISSUE THREAD

**Issue by mikerite** [rust-lang/rust-clippy#7781]
I tried this code:

```rust
#![warn(clippy::equatable_if_let)]

macro_rules! m1 {
    (x) => { "abc" };
    (y) => { "def" };
}

fn main() {
    if let m1!(x) = "abc" { 
        println!("OK");
    }
}
```

I expected to see this happen: I would get recommendation `if "abc" == m1!(x)`

Instead, this happened:
```
warning: this pattern matching can be expressed using equality
 --> src/main.rs:9:8
  |
9 |     if let m1!(x) = "abc" { 
  |        ^^^^^^^^^^^^^^^^^^ help: try: `"abc" == "abc"`
  |
```
The macro call is expanded. This code isn't equivalent depending on the macro.

### Meta
Run on clippy master - 01ea06acb730fea7cea41ff3dba414f41449af2f

@rustbot label +L-suggestion
